DEV-INTRO.md

```diff
@@ -1,59 +1,166 @@
+
+- **lib** holds all the R packages for the specific version of PEcAn and R. This folder will be shared amongst all other containers, and will contain the compiled PEcAn code.
+- **pecan** this holds all the data, such as workflows and any downloaded data.
+- **portainer** if you enabled the portainer service this folder is used to hold persistent data for this service
+- **postgres** holds the actual database data. If you want to backup the database, you can stop the postgres container, zip up the folder.
+- **rabbitmq** holds persistent information of the message broker (rabbitmq). 
+- **traefik** holds persisent data for the web proxy, that directs incoming traffic to the correct container.
+
+These folders will hold all the persistent data for each of the respective containers and can grow. For example the postgres database is multiple GB. The pecan folder will hold all data produced by the workflows, including any downloaded data, and can grow to many giga bytes.
+
+#### postgresql database
+
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
+
+Once the database has finished starting up we will initialize the database using: `docker run --rm --network pecan_pecan pecan/db`. Once that is done we create two users for BETY:
 
 ```
-library(devtools)
-install_github("pecan", "PEcAnProject")
+# guest user
+docker-compose run --rm bety user guestuser guestuser "Guest User" guestuser@example.com 4 4
+
+# example user
+docker-compose run --rm bety user carya illinois "Carya Demo User" carya@example.com 1 1
 ```
 
-#### "Makefile"
+#### copy web config file
+
+The `docker-compose.dev.yaml` file has a section that will eanble editing the web application. This is by default commented out. If you want to uncoment it you will need to first copy the config.php from the docker/web folder. You can do this using `cp docker/web/config.docker.php web/config.php`.
+
+#### copy R packages
+
+Next copy the R packages from a container to your local machine as the `volumes/lib` folder. This is not really needed, but will speed up the process of the first compilation. Later we will put our newly compiled code here as well. 
+
+You can copy all the data using `docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -a /usr/local/lib/R/site-library/. /rlib/`. This will copy all compiled packages to your local machine.
```

Comment:
I think somewhere around here first timers need to run `docker run -ti --rm --network pecan_pecan --volume pecan_pecan:/data --env FQDN=docker pecan/data:develop` to get core db files

Reply:
I ran this command and also `docker pull pecan/db`, but it looks like I still don't have that met file with id = 5000000005. I get this error from the updated `docker.sipnet.xml`: 
```
2020-05-26 15:38:26 ERROR  [PEcAn.data.atmosphere::met.process] :
   Cannot process met without source information
Error in paste0("registration/register.", met, ".xml") :
  object 'met' not found
Calls: <Anonymous> -> <Anonymous> -> system.file -> file.path -> paste0
```
Somewhat relatedly, when trying to log into the localhost Betydb, using carya as the login and illinois as the password doesn't work for me. Is login info something else?

Reply:
If `docker-compose run --rm bety user carya illinois "Carya Demo User" carya@example.com 1 1 ` returns success, that is the user-password combination yes..but this is one of the problematic lines, i.e. you need to do `docker-compose -f docker-compose.yml -f docker-compose.dev.yml run --rm bety user carya illinois "Carya Demo User" carya@example.com 1 1` if you don't have dev.yml as override.yml

In general, I seem to be missing all the records with 5000****  (assuming that's Winsconsin), I also don't have` http://localhost:8000/bety/formats/5000000002` for example which is the     `AMERIFLUX_BASE_HH` format

Reply:
Thanks for suggesting this @istfer. I can't seem to get the bety user to work. I'm either getting the same traefik configuration error as above, or when I rename `docker-compose.dev.yml` to `docker-compose.override.yml`, it returns `/home/bety/docker/entrypoint.sh: line 86: exec: user: not found`. I'm not sure what I'm doing wrong?

Reply:
Not sure, but if you are willing to start from scratch I have listed the steps here that worked for me (I'm working on ubuntu tho) https://github.com/robkooper/pecan/pull/5/files

Reply:
but if you try it you need to run the code below after the 5th step since this branch is not merged yet:
```
git remote add robkooper https://github.com/robkooper/pecan.gitgit 
fetch robkooper
git checkout --track robkooper/docker-develop
```

Reply:
the record for the met data is from Wisconsin (5000000002). Not sure what the easiest way to clean this up. You can do the following (potentially not 100% tested).

```
docker-compose down
docker pull pecan/db
# not quite remove the old data, but effectively the same
mv volumes volumes.old
```

Now start the from "First time setup". This should load the data with the input record for [5000000002](http://localhost:8000/bety/inputs/5000000005).

Reply:
Don't know about @KristinaRiemer but this still don't give me input or format record with id `5000000002`

Reply:
That id didn't work for me a few days ago, I'll try these instructions again and let y'all know.

Reply:
I went through @robkooper instructions above by running `docker-compose down`, `docker pull pecan/db`, and `mv volumes volumes.old` and then going through the instructions from "First time setup". I still don't see any data for 5000000002 or 5000000005. Do these come in with the "load example data" step? That's the only thing I couldn't get to work due to the FQDN error.

Reply:
I tried `docker run --rm --network pecan_pecan --volume pecan_pecan:/data --env FQDN=docker pecan/data:develop` instead, but still no 5000000002